OpenRA.Mods.Common/Traits/World/ColorPickerManager.cs

```diff
@@ -179,8 +127,100 @@ Color MakeValid(float hue, float sat, MersenneTwister random, List<(float R, flo
 
 			// Failed to find a solution within a reasonable time: return a random color without any validation
 			onError?.Invoke(InvalidPlayerColor);
-			return Color.FromAhsv(random.NextFloat(), float2.Lerp(HsvSaturationRange[0], HsvSaturationRange[1], random.NextFloat()), V);
+			var randomSat = float2.Lerp(HsvSaturationRange[0], HsvSaturationRange[1], random.NextFloat());
+			var randomVal = float2.Lerp(HsvSaturationRange[0], HsvSaturationRange[1], random.NextFloat());
+			return Color.FromAhsv(random.NextFloat(), randomSat, randomVal);
+		}
+
+		#region IColorPickerManagerInfo
+
+		public event Action<Color> OnColorPickerColorUpdate;
+
+		(float hMin, float hMax) IColorPickerManagerInfo.GetHueRange { get; }
```

Comment:
This is never assigned to?

Reply:
Yes, this is just the implementation of the interface. In our setup we use hue slider and don't limit hue

Reply:
tho I suppose it doesn't make any sense to have this. If other implementations need it they will extend the interface

removed

Reply:
Well actually not needing it is an implementation detail specific to our implementation. The interface makes sense to have it, being a supposedly generic one. If we don't need it, then I think something like this would be best
```cs
// Our implementation doesn't need or use this because we get the hue from the HueSliderWidget.
(float hMin, float hMax) IColorPickerManagerInfo.GetHueRange => (0, 0)`;
```

Reply:
I've already made up my mind. It makes 0 sense to have it.

* We don't use it
* The color manager doesn't use it
* Hue range restrictions look completely different. Restricting a circle between 2 values does not make any sense

Reply:
It doesn't make sense to have a hue range, as it does not have a natural min or max value (its circular).